src/netex_idf/offers.rs

```diff
@@ -180,6 +180,18 @@ fn parse_common(_common: &Element) -> Result<()> {
     Ok(())
 }
 
+fn parse_service_journey_patterns<'a, I>(sjp_elements: I) -> HashMap<String, &'a Element>
+where
+    I: Iterator<Item = &'a Element>,
+{
+    sjp_elements
+        .filter_map(|sjp_element| {
+            let id: String = sjp_element.attribute("id")?;
+            Some((id, sjp_element))
```

Comment:
why you don't return `Some((route_ref, jp_id.clone()))` directly as you do in `enhance_with_object_code`?

Reply:
Because as I said [here](https://github.com/CanalTP/transit_model/pull/441#discussion_r340664284), I'm going to need more than just the ID for the coming PR, so I'm returning the full `minidom::Element` which contains all the information.